### PR TITLE
Gutenboarding: rollback to v5

### LINKED ISSUE #44980: Gutenboarding: roll back to v5
We've now re-tested v2 and v5 mechanics in Gutenboarding. Let's go back to v5 as our base for further iteration with the following changes:

- [x] Add domains back as a mandatory step after site name input
- [x] Hide header domain nudge while at domains step
  - In V5 domain nudge is hidden while plans grid is open, but I don't see issue just leaving it there.
- [x] Hide vertical input for everyone (perhaps hide behind a feature flag for now?)
  - [x] Remove test definition
- [x] Update the a/b tests slugs too (both the gutenboarding main one and the existing users test)
- [ ] We can close two click plans PR after all :-) https://github.com/Automattic/wp-calypso/pull/44775
- Keep small bugfixes and improvements to the header nudge that we did while shuffling v2/v5 around previously

### PR DESCRIPTION
**Pull request by razvanpapadopol**
#### Changes proposed in this Pull Request

_Copied from #44980_

- [x] Add domains back as a mandatory step after site name input
- [x] Hide header domain nudge while at domains step
  - In V5 domain nudge is hidden while plans grid is open, but I don't see issue just leaving it there.
- [x] Hide vertical input for everyone (perhaps hide behind a feature flag for now?)
  - [x] Remove test definition
- [x] Update the a/b tests slugs too (both the gutenboarding main one and the existing users test)

#### Testing instructions

1. Go to `/new/`, you shouldn't see vertical input. 
2. Fill your site name, the button text should be "Choose a domain", not "Choose a design".
3. In Domain step, the domain nudge shouldn't show.
4. Open the plans modal by clicking "View Plans", the domain button on the top left should remain visible.

Fixes #44980

### CODE REVIEW

File: client/landing/gutenboarding/onboarding-block/acquire-intent/index.tsx [hunk 1]
```diff
@@ -74,7 +74,7 @@ const AcquireIntent: React.FunctionComponent = () => {
 		<SiteTitle inputRef={ siteTitleRef } onSubmit={ handleSiteTitleSubmit } />
 	);
 	const nextStepButton = hasSiteTitle ? (
-		<NextButton onClick={ handleSiteTitleSubmit }>{ __( 'Choose a design' ) }</NextButton>
+		<NextButton onClick={ handleSiteTitleSubmit }>{ __( 'Choose a domain' ) }</NextButton>
```
Comment (razvanpapadopol) [inline client/landing/gutenboarding/onboarding-block/acquire-intent/index.tsx:77]: Let's go with _Continue_ / _Skip for now_ as illustrated in designs for v5.

**Comment by simison** [comment]
Any of the hooks that were added in these PRs that needs cleaning out?
https://github.com/Automattic/wp-calypso/pull/44837/files
https://github.com/Automattic/wp-calypso/pull/44822/files

**Comment by simison** [comment]
Added e2e label — I suspect we might need some changes there.

File: client/lib/abtest/active-tests.js [hunk 2]
```diff
@@ -224,14 +224,6 @@ export default {
...
 		countryCodeTargets: [ 'US', 'CA' ],
 	},
-	verticalsOnGutenboarding: {
-		datestamp: '20200811',
-		variations: {
-			verticals: 50,
-			noverticals: 50,
-		},
-		defaultVariation: 'noverticals',
-	},
```
Comment (razvanpapadopol) [inline client/lib/abtest/active-tests.js:234]: Together with removing the test definition, we can probably revert also its usage: https://github.com/Automattic/wp-calypso/pull/44256/files#diff-46ce4cd8580489aebbb3447a87c3cb57

**Pull request merged**